tests/test_annual_time.py

```diff
@@ -0,0 +1,140 @@
+    yield AnnualTimeDimensionConfig(
+        AnnualTimeDimensionModel(
+            dimension_type=DimensionType.TIME,
+            class_name="AnnualTime",
+            module="dsgrid.dimension.standard",
+            name="annual_time",
+            description="test annual time",
+            display_name="Annual",
+            ranges=[
+                TimeRangeModel(start="2010", end="2020"),
+            ],
+            str_format="%Y",
+            measurement_type=MeasurementType.TOTAL,
+            include_leap_day=True,
+        )
+    )
+
+
+@pytest.fixture
+def date_time_dimension():
+    yield DateTimeDimensionConfig(
+        DateTimeDimensionModel(
+            dimension_type=DimensionType.TIME,
+            class_name="Time",
+            module="dsgrid.dimension.standard",
+            frequency="P0DT1H",
+            timezone=TimeZone.EST,
+            name="datetime",
+            description="example date time",
+            display_name="time_est",
+            ranges=[
+                TimeRangeModel(start="2012-02-01 00:00:00", end="2012-02-07 23:00:00"),
+            ],
+            str_format="%Y-%m-%d %H:%M:%S",
+            time_interval_type=TimeIntervalType.PERIOD_BEGINNING,
+            measurement_type=MeasurementType.TOTAL,
+        )
+    )
+
+
+def test_map_annual_time_measured_to_datetime(
+    annual_dataframe, annual_time_dimension, date_time_dimension
+):
+    annual_time_dimension.model.measurement_type = MeasurementType.MEASURED
+    model_years = [2020, 2024]
+    df = annual_time_dimension.map_annual_time_measured_to_datetime(
+        annual_dataframe, date_time_dimension, model_years
+    )
+    assert df.count() == 168
+    values = df.select("electricity_sales").distinct().collect()
+    assert len(values) == 1
+    assert (
+        values[0].electricity_sales
+        == annual_dataframe.filter("time_year == 2020").collect()[0].electricity_sales
+    )
+    mapped_model_years = (
+        df.withColumn("year", F.year(df.timestamp)).select("year").distinct().collect()
+    )
+    assert len(mapped_model_years) == 1
+    assert mapped_model_years[0].year == 2020
+
+
+def test_map_annual_time_total_to_datetime(
+    annual_dataframe, annual_time_dimension, date_time_dimension
+):
+    annual_time_dimension.model.measurement_type = MeasurementType.TOTAL
+    model_years = [2020, 2024]
+    value_columns = ["electricity_sales"]
+    df = annual_time_dimension.map_annual_total_to_datetime(
+        annual_dataframe, date_time_dimension, model_years, value_columns
+    )
+    assert df.count() == 168
+    values = df.select("electricity_sales").distinct().collect()
+    assert len(values) == 1
+    num_intervals_2020 = (
+        366 * 24 / (date_time_dimension.model.frequency.total_seconds() / (60 * 60))
+    )
+    assert (
+        values[0].electricity_sales
+        == annual_dataframe.filter("time_year == 2020").collect()[0].electricity_sales
+        / num_intervals_2020
+    )
```

Comment:
Nit: I guess this is passing as is, but we could use [np.isclose](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html) for comparing floats

Reply:
I see that in another file we use math.isclose

Reply:
I think that the math here should be identical. Unlike cases where we perform the math in slightly different ways. Interesting note in numpy.isclose docs:

```
Unlike the built-in `math.isclose`, the above equation is not symmetric
in `a` and `b` -- it assumes `b` is the reference value -- so that
`isclose(a, b)` might be different from `isclose(b, a)`.
```